[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/miguepoloc/toma-decisiones-mcda/blob/main/02_topsis_cacao.ipynb)

# TOPSIS, caso cacao

Mismo caso (Sierra Nevada de Santa Marta, 4 zonas, 4 variables). Contenido completo en la Sesión 3 (TOPSIS/VIKOR) del curso.

In [1]:
!pip install -q pyDecision


[notice] A new release of pip is available: 26.1.2 -> 26.2.1
[notice] To update, run: pip install --upgrade pip


In [2]:
import numpy as np
from pyDecision.algorithm import topsis_method

zonas = ["Bonda", "Guachaca", "San Pedro", "Palmor"]
dataset = np.array([
    [27, 78, 6.2, 0.9],   # Bonda
    [26, 85, 5.8, 1.5],   # Guachaca
    [24, 82, 6.5, 0.7],   # San Pedro
    [22, 88, 5.5, 1.2],   # Palmor
])
pesos = [0.25, 0.30, 0.25, 0.20]
tipo = ['max', 'min', 'max', 'min']  # Temp, Humedad, pH, CE

## Cálculo y ranking

In [3]:
resultado = topsis_method(dataset, pesos, tipo, graph=False, verbose=False)

print("Ranking final TOPSIS:")
for z, c in sorted(zip(zonas, resultado), key=lambda x: -x[1]):
    print(f"  {z:12s} Ci = {c:.3f}")

Ranking final TOPSIS:
  San Pedro    Ci = 0.820
  Bonda        Ci = 0.771
  Palmor       Ci = 0.316
  Guachaca     Ci = 0.226


**Resultado esperado** (coincide con lo publicado en la Sesión 3 del curso): 1º San Pedro (0.820) · 2º Bonda (0.771) · 3º Palmor (0.316) · 4º Guachaca (0.226).